plugins/automelt.cpp

```diff
@@ -625,48 +626,67 @@ static void automelt_printStatus(color_ostream &out) {
         if (!isStockpile(stockpile)) continue;
-        name_width = std::max(name_width, (int)stockpile->name.size());
+        if (stockpile->name.empty()) {
+            string stock_name = "Stockpile #" + int_to_string(stockpile->stockpile_number);
+            name_width = std::max(name_width, (int)(stock_name.size()));
+        } else {
+            name_width = std::max(name_width, (int)stockpile->name.size());
+        }
     }
     name_width = -name_width;
 
-    const char *fmt = "%*s  %4s  %4s  %5s  %5s\n";
-    out.print(fmt, name_width, "name", " id ", "monitored", "items", "marked");
-    out.print(fmt, name_width, "----", "----", "---------", "-----", "------");
+    const char *fmt = "%*s  %9s  %5s  %6s\n";
+    out.print(fmt, name_width, "name", "monitored", "items", "marked");
+    out.print(fmt, name_width, "----", "---------", "-----", "------");
 
     for (auto &stockpile : world->buildings.other.STOCKPILE) {
         if (!isStockpile(stockpile)) continue;
         bool monitored = false;
         int32_t item_count = 0;
         int32_t marked_item_count = 0;
-        if (watched_stockpiles_indices.count(stockpile->id)) {
-            auto &c = watched_stockpiles[watched_stockpiles_indices[stockpile->id]];
+        if (watched_stockpiles.count(stockpile->id)) {
+            auto &c = watched_stockpiles[stockpile->id];
             monitored = get_config_bool(c, STOCKPILE_CONFIG_MONITORED);
             int id = get_config_val(c, STOCKPILE_CONFIG_ID);
             item_count = item_count_piles[id];
             marked_item_count = premarked_item_count_piles[id];
+        }
 
+        int32_t stockpile_number = stockpile->stockpile_number;
+        if (stockpile->name.empty()) {
+            string stock_name = "Stockpile #" + int_to_string(stockpile->stockpile_number);
+            out.print(fmt, name_width, stock_name, monitored ? "[x]": "[ ]",
+                        int_to_string(item_count).c_str(), int_to_string(marked_item_count).c_str());
+        } else {
+            out.print(fmt, name_width, stockpile->name.c_str(), monitored ? "[x]": "[ ]",
+                        int_to_string(item_count).c_str(), int_to_string(marked_item_count).c_str());
         }
 
-        out.print(fmt, name_width, stockpile->name.c_str(), int_to_string(stockpile->id).c_str(), monitored ? "[x]": "[ ]",
-                    int_to_string(item_count).c_str(), int_to_string(marked_item_count).c_str());
+
     }
     DEBUG(status,out).print("exiting automelt_printStatus\n");
 
 }
 
 static void automelt_setStockpileConfig(color_ostream &out, int id, bool monitored) {
-    DEBUG(status,out).print("entering automelt_setStockpileConfig\n");
+    DEBUG(status,out).print("entering automelt_setStockpileConfig for id=%d and monitored=%d\n", id, monitored);
     validate_stockpile_configs(out);
-    bool isInvalidStockpile = !df::building::find(id);
+    bool isInvalidStockpile = !df::building::find(id) || !isStockpile(df::building::find(id));
```

Comment:
find is O(log n), which is not free. I'd cache the return value instead of running the search twice.